fs/fuse/fuse_dlm_cache.h

```diff
@@ -0,0 +1,64 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+/*
+ * FUSE page cache lock implementation
+ */
+
+#ifndef _FS_FUSE_DLM_CACHE_H
+#define _FS_FUSE_DLM_CACHE_H
+
+#include <linux/types.h>
+#include <linux/rbtree.h>
+#include <linux/list.h>
+#include <linux/rwsem.h>
+
+/* Lock modes for page ranges */
+enum fuse_page_lock_mode { FUSE_PAGE_LOCK_READ, FUSE_PAGE_LOCK_WRITE };
+
+/* A range of pages with a lock */
+struct fuse_dlm_range {
```

Comment:
Do we actually need to expose it in the header or can we move it to fuse_dlm_cache.c?